.gitlab-ci.yml

```diff
@@ -179,6 +179,40 @@ requirements_json_test:
     REQUIREMENTS_BLOCK_JSON_PATH: ".gitlab/requirements_block.json"
     REQUIREMENTS_ALLOW_JSON_PATH: ".gitlab/requirements_allow.json"
 
+detect_circular_imports:
+  script:
+    - cd ..
+    - python -m pip install -r dd-trace-py/scripts/import-analysis/requirements-cycles.txt
+    - cp dd-trace-py/scripts/import-analysis/cycles.py .
+    - python dd-trace-py/scripts/import-analysis/cycles.py analyze cycles-pr.json
+    - cat cycles-pr.json
+    - cd dd-trace-py && git checkout main && cd ..
```

Comment:
should we only run this job on non-main pipelines?

or else, we could have the runs on `main` fill a cache with it's base data, and the each PR pipeline could just use the last cached value from main.